src/cascadia/TerminalSettingsEditor/Profiles_Advanced.cpp

```diff
@@ -10,22 +10,103 @@
 #include <LibraryResources.h>
 #include "..\WinRTUtils\inc\Utils.h"
 
+using namespace winrt::Windows::Foundation;
+using namespace winrt::Windows::UI::Xaml;
+using namespace winrt::Windows::UI::Xaml::Controls;
 using namespace winrt::Windows::UI::Xaml::Navigation;
 
 namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
 {
     Profiles_Advanced::Profiles_Advanced()
     {
         InitializeComponent();
+
+        Automation::AutomationProperties::SetName(AddBellSoundButton(), RS_(L"Profile_AddBellSound/Text"));
     }
 
     void Profiles_Advanced::OnNavigatedTo(const NavigationEventArgs& e)
     {
-        _Profile = e.Parameter().as<Editor::ProfileViewModel>();
+        const auto args = e.Parameter().as<Editor::NavigateToProfileArgs>();
+        _Profile = args.Profile();
+        _windowRoot = args.WindowRoot();
     }
 
     void Profiles_Advanced::OnNavigatedFrom(const NavigationEventArgs& /*e*/)
     {
         _ViewModelChangedRevoker.revoke();
     }
+
+    safe_void_coroutine Profiles_Advanced::BellSoundAudioPreview_Click(const IInspectable& sender, const RoutedEventArgs& /*e*/)
+    try
+    {
+        const auto path = sender.as<Button>().Tag().as<Editor::BellSoundViewModel>().Path();
+        if (path.empty())
+        {
+            co_return;
+        }
+        winrt::hstring soundPath{ wil::ExpandEnvironmentStringsW<std::wstring>(path.c_str()) };
+        winrt::Windows::Foundation::Uri uri{ soundPath };
+
+        auto weakThis{ get_weak() };
+        co_await wil::resume_foreground(Dispatcher());
```

Comment:
Since this comes in off a `Click` event handler, I think it is guaranteed to already be on the UI thread!

Reply:
if you take advantage of that guarantee, you can drop all the `strongThis` stuff too